pkgs/by-name/fr/frequest/package.nix

```diff
@@ -0,0 +1,69 @@
+{ lib, stdenv, fetchFromGitHub, qt5 }:
+
+let
+  version = "1.2a";
+
+  src = fetchFromGitHub {
+    owner = "fabiobento512";
+    repo = "FRequest";
+    rev = "v${version}";
+    hash = "sha256-fdn3MK5GWBOhJjpMtRaytO9EsVzz6KJknDhqWtAyXCc=";
+  };
+
+  commonLibsSrc = fetchFromGitHub {
+    owner = "fabiobento512";
+    name = "CommonLibs";
+    repo = "CommonLibs";
+    rev = "d3906931bb06ddf4194ff711a59e1dcff80fa82f";
+    hash = "sha256-iLJJ95yJ+VjNPuk8fNEDvYBI0db0rcfJF12a9azGv1Y=";
+  };
+
+  commonUtilsSrc = fetchFromGitHub {
+    owner = "fabiobento512";
+    name = "CommonUtils";
+    repo = "CommonUtils";
+    rev = "53970984f6538d78350be1b9426032bcb5bcf818";
+    hash = "sha256-nRv9DriSOuAiWhy+KkOVNEz5oSgNNNJZqk8sNwgbx8U=";
+  };
+
+in stdenv.mkDerivation {
+
+  pname = "frequest";
+  inherit version;
+
+  srcs = [
+    src
+    commonLibsSrc
+    commonUtilsSrc
+  ];
+  sourceRoot = src.name;
```

Comment:
This is an interesting pattern I never seen before. Could you explain a bit how it works? in a comment please. What's not clear to me, is how upstream's qmake scripts find the sources from the shell variable `$srcs`...

Also, I think you don't need the `let in` block, and you can put all the srcs in the list right at it. But that's a styling recommendation.

Reply:
@doronbehar , I've added a comment to explain it a bit, hope it's clear.
There's not much magic in the implementation, the upstream simply depends on `../CommonsLibs` and `CommonUtils`. That's why the `name` of the `srcs` and the `sourceRoot` are critical to this package.